test/Identity.IntegrationTest/Endpoints/IdentityServerTwoFactorTests.cs

```diff
@@ -1,143 +1,493 @@
-﻿using System.Text.Json;
+﻿using System.Security.Claims;
+using System.Text.Json;
 using Bit.Core.AdminConsole.Entities;
+using Bit.Core.Auth.Entities;
 using Bit.Core.Auth.Enums;
+using Bit.Core.Auth.Models.Data;
+using Bit.Core.Auth.Repositories;
 using Bit.Core.Entities;
 using Bit.Core.Enums;
+using Bit.Core.Models.Data;
 using Bit.Core.Repositories;
 using Bit.Core.Services;
+using Bit.Core.Utilities;
 using Bit.Identity.Models.Request.Accounts;
 using Bit.IntegrationTestCommon.Factories;
 using Bit.Test.Common.AutoFixture.Attributes;
 using Bit.Test.Common.Helpers;
-using Microsoft.AspNetCore.TestHost;
+using Duende.IdentityServer.Models;
+using Duende.IdentityServer.Stores;
+using IdentityModel;
+using LinqToDB;
+using NSubstitute;
 using Xunit;
 
+// #nullable enable
+
 namespace Bit.Identity.IntegrationTest.Endpoints;
 
 public class IdentityServerTwoFactorTests : IClassFixture<IdentityApplicationFactory>
 {
+    const string _organizationTwoFactor = """{"6":{"Enabled":true,"MetaData":{"IKey":"DIEFB13LB49IEB3459N2","SKey":"0ZnsZHav0KcNPBZTS6EOUwqLPoB0sfMd5aJeWExQ","Host":"api-example.duosecurity.com"}}}""";
+    const string _testEmail = "test+2farequired@email.com";
+    const string _testPassword = "master_password_hash";
+    const string _userEmailTwoFactor = """{"1": { "Enabled": true, "MetaData": { "Email": "test+2farequired@email.com"}}}""";
+
     private readonly IdentityApplicationFactory _factory;
-    private readonly IUserRepository _userRepository;
-    private readonly IUserService _userService;
 
     public IdentityServerTwoFactorTests(IdentityApplicationFactory factory)
     {
         _factory = factory;
-        _userRepository = _factory.GetService<IUserRepository>();
-        _userService = _factory.GetService<IUserService>();
     }
 
-    [Theory, BitAutoData]
-    public async Task TokenEndpoint_UserTwoFactorRequired_NoTwoFactorProvided_Fails(string deviceId)
+    [Fact]
+    public async Task TokenEndpoint_GrantTypePassword_UserTwoFactorRequired_NoTwoFactorProvided_Fails()
     {
         // Arrange
-        var username = "test+2farequired@email.com";
-        var twoFactor = """{"1": { "Enabled": true, "MetaData": { "Email": "test+2farequired@email.com"}}}""";
+        await CreateUserAsync(_factory, _testEmail, _userEmailTwoFactor);
```

Comment:
I think we might mis-use fixtures in some places since the primary benefit to fixtures is they share context, but it's the sharing context that is causing the problem. 

Most of our testing is done atomically outside the context of any other tests in the class.

I'll take a look to see what I can tighten up to make the class cleaner. Although I'm not sure we'll be able to fully utilize the shared context of the fixture.